tests/test_enterprise/api/test_filters.py

```diff
@@ -13,30 +13,76 @@
 
 from django.conf import settings
 
-from test_utils import TEST_EMAIL, TEST_PASSWORD, TEST_USERNAME, APITest, factories
+from test_utils import FAKE_UUIDS, TEST_EMAIL, TEST_PASSWORD, TEST_USERNAME, APITest, factories
 
 
 @ddt.ddt
 @mark.django_db
-class TestEnterpriseCustomerUserFilterBackend(APITest):
+class TestUserFilterBackend(APITest):
     """
-    Tests for enterprise API throttling.
+    Test suite for the ``UserFilterBackend`` filter.
     """
 
-    def create_user(self, username=TEST_USERNAME, password=TEST_PASSWORD, **kwargs):
+    def setUp(self):
+        self.create_user(id=1)
+        self.enterprise_customer = factories.EnterpriseCustomerFactory(uuid=FAKE_UUIDS[0])
+        self.enterprise_customer_user = factories.EnterpriseCustomerUserFactory(
+            enterprise_customer=self.enterprise_customer,
+            user_id=self.user.id,
+        )
+        self.client = APIClient()
+        self.client.login(username=TEST_USERNAME, password=TEST_PASSWORD)
+
+    @ddt.data(
+        (False, 1),
+        (True, 2),
+    )
+    @ddt.unpack
+    def test_filter_for_list(self, is_staff, expected_data_length):
         """
-        Create a test user and set its password.
+        Filter objects based off of user ID -- show all for staff, but only the authenticated user himself otherwise.
         """
-        self.user = factories.UserFactory(username=username, is_active=True, **kwargs)
-        self.user.set_password(password)  # pylint: disable=no-member
-        self.user.save()  # pylint: disable=no-member
+        self.user.is_staff = is_staff
+        self.user.save()
+        factories.EnterpriseCustomerFactory()
+        response = self.client.get(settings.TEST_SERVER + reverse('enterprise-customer-list'))
```

Comment:
Hmm, not really, no. This just unit tests the filter, which has the same set of logic for all endpoints, and that's what we're really testing, so what endpoint is used is irrelevant as long as that endpoint uses this filter.

Reply:
@UmanShahzad OK, let's add a note about that to the docstring:

```
Filter objects based off of user ID -- show all for staff, but only the authenticated user himself otherwise.

Note that the choice of endpoint below is arbitrary.
This test could use any endpoint that makes use of the ``UserFilterBackend`` filter.
```

Reply:
:+1: